src/grammar.coffee

```diff
@@ -47,9 +47,14 @@ o = (patternString, action, options) ->
     getAddDataToNodeFunctionString = (first, last) ->
       "yy.addDataToNode(yy, @#{first}#{if last then ", @#{last}" else ''})"
 
+    returnsLoc = /^LOC/.test action
```

Comment:
The existing grammar-rule function always wrapped the return value (the newly constructed node class object) in a call to `LOC()` (the location-data-attaching function)

But we now want to have a grammar rule that uses a non-default `LOC()` for the returned node class object (see below). So here introduce extra logic that only wraps the return value in `LOC()` if the grammar rule isn't already explicitly doing so